xds/csds/csds_e2e_test.go

```diff
@@ -304,91 +333,91 @@ func (s) TestCSDS(t *testing.T) {
-		config := makeGenericXdsConfig("type.googleapis.com/envoy.config.cluster.v3.Cluster", cdsTargets[i], "2", v3adminpb.ClientResourceStatus_ACKED, clusterAnys[i])
-		if i == nackResourceIdx {
-			config.VersionInfo = "1"
-			config.ClientStatus = v3adminpb.ClientResourceStatus_NACKED
-			config.ErrorState = &v3adminpb.UpdateFailureState{VersionInfo: "2"}
-		}
-		want = append(want, config)
-	}
-	for i := range edsTargets {
-		config := makeGenericXdsConfig("type.googleapis.com/envoy.config.endpoint.v3.ClusterLoadAssignment", edsTargets[i], "2", v3adminpb.ClientResourceStatus_ACKED, endpointAnys[i])
-		if i == nackResourceIdx {
-			config.VersionInfo = "1"
-			config.ClientStatus = v3adminpb.ClientResourceStatus_NACKED
-			config.ErrorState = &v3adminpb.UpdateFailureState{VersionInfo: "2"}
-		}
-		want = append(want, config)
-	}
-	for {
-		if err := ctx.Err(); err != nil {
-			t.Fatalf("Timeout when waiting for resources in \"NACKed\" state: %v", err)
-		}
-		err := checkClientStatusResponse(stream, want)
-		if err == nil {
-			break
-		}
-		time.Sleep(time.Millisecond * 100)
+	wantConfigs = []*v3statuspb.ClientConfig_GenericXdsConfig{
+		makeGenericXdsConfig("type.googleapis.com/envoy.config.cluster.v3.Cluster", cdsTargets[0], "1", v3adminpb.ClientResourceStatus_NACKED, clusterAnys[0], &v3adminpb.UpdateFailureState{VersionInfo: "2"}),
+		makeGenericXdsConfig("type.googleapis.com/envoy.config.cluster.v3.Cluster", cdsTargets[1], "2", v3adminpb.ClientResourceStatus_ACKED, clusterAnys[1], nil),
+		makeGenericXdsConfig("type.googleapis.com/envoy.config.endpoint.v3.ClusterLoadAssignment", edsTargets[0], "1", v3adminpb.ClientResourceStatus_NACKED, endpointAnys[0], &v3adminpb.UpdateFailureState{VersionInfo: "2"}),
+		makeGenericXdsConfig("type.googleapis.com/envoy.config.endpoint.v3.ClusterLoadAssignment", edsTargets[1], "2", v3adminpb.ClientResourceStatus_ACKED, endpointAnys[1], nil),
+		makeGenericXdsConfig("type.googleapis.com/envoy.config.listener.v3.Listener", ldsTargets[0], "1", v3adminpb.ClientResourceStatus_NACKED, listenerAnys[0], &v3adminpb.UpdateFailureState{VersionInfo: "2"}),
+		makeGenericXdsConfig("type.googleapis.com/envoy.config.listener.v3.Listener", ldsTargets[1], "2", v3adminpb.ClientResourceStatus_ACKED, listenerAnys[1], nil),
+		makeGenericXdsConfig("type.googleapis.com/envoy.config.route.v3.RouteConfiguration", rdsTargets[0], "1", v3adminpb.ClientResourceStatus_NACKED, routeAnys[0], &v3adminpb.UpdateFailureState{VersionInfo: "2"}),
+		makeGenericXdsConfig("type.googleapis.com/envoy.config.route.v3.RouteConfiguration", rdsTargets[1], "2", v3adminpb.ClientResourceStatus_ACKED, routeAnys[1], nil),
+	}
+	wantResp = &v3statuspb.ClientStatusResponse{
+		Config: []*v3statuspb.ClientConfig{
+			{
+				Node:              wantNode,
+				GenericXdsConfigs: wantConfigs,
+				ClientScope:       xdsClient1Name,
+			},
+			{
+				Node:              wantNode,
+				GenericXdsConfigs: wantConfigs,
+				ClientScope:       xdsClient2Name,
+			},
+		},
```

Comment:
This adds testing scope for the numerous xDS Clients and client scope. However, they point to the same resources. Should we test the clients pointing to different resources to make sure all that works correctly? Also another dimension alongside different resources is some resources in one client being in an error state while others not, alongside just a raw different state, which I also think is worth testing.

Reply:
I have tests in the `xdsclient` package which creates xDS clients talking to different servers. Adding the same tests here, I feel does not add much value, since this already tests the CSDS functionality thoroughly.

Reply:
Hmmm fair point, this tests the logic for errors/different CSDS operaitons.

Reply:
Just make sure maybe manually in the future CSDS looks right with multiple clients that point to different servers that have different resource states/errors for those resources.